File: LittleProjects/Scala/AkkaQuillSQLCrud/src/main/scala/akka/quill/sql/crud/service/Service.scala
package akka.quill.sql.crud.service

trait Service[A, ID] {
  def findOne(id: ID): Either[Throwable, Option[A]]

  def findAll(): Either[Throwable, Seq[A]]

  def save(a: A): Either[Throwable, A]

  def update(a: A): Either[Throwable, A]

  def deleteOne(id: ID): Either[Throwable, Long]

  def deleteAll(): Either[Throwable, Long]
}

File: LittleProjects/Java/GuiceHibernatePostgresCrud/src/main/java/guice/hibernate/postgres/crud/service/Service.java
package guice.hibernate.postgres.crud.service;

import java.util.Optional;

public interface Service<A, Id> {
    Optional<A> findOne(Id id);

    Iterable<A> findAll();

    A save(A a);

    A update(A a);

    boolean deleteOne(Id id);

    long deleteAll();
}
